File: python/knuth_shuffle/knuth_shuffle.py
import random

def knuthShuffle(ary):
  aryLen = len(ary)
  for i in range(1, aryLen+1):
    j = aryLen - i
    if j > 0:
      swapI = random.randint(0, j-1)
    else:
      swapI = 0
    temp = ary[swapI]
    ary[swapI] = ary[j]
    ary[j] = temp
  pass

ary = [0,1,2,3,4,5,6,7,8,9]
print ary
knuthShuffle(ary)
print ary

File: javascript/knuth_shuffle/knuth_shuffle.js
var aryLen = 10;
var ary = new Array(10).fill(true).map((e,i) => i);
console.log(ary);

function knuthShuffle(ary) {
  for (var i = ary.length - 1; i > 0; i--) {
    var swap = Math.floor(Math.random() * i);
    var temp = ary[swap];
    ary[swap] = ary[i];
    ary[i] = temp;
  }
}

knuthShuffle(ary);
console.log(ary);